### DISCUSSION TITLE
Clarification of FieldNames

### DISCUSSION THREAD

**Discussion by MFK2021** [awslabs/gluonts#1770] [Q&A] [answered]
Hi everyone, in order to convert my csv file to a format which can be used by the GluonTS library, I will use FieldNames. 

FEAT_STATIC_CAT , FEAT_STATIC_REAL, FEAT_DYNAMIC_CAT  and FEAT_DYNAMIC_REAL are pretty straight forward and easy to understand. But what is the meaning of the following FieldNames and how do they differ to the previous ones? : 
- PAST_FEAT_DYNAMIC_REAL ? 
- FEAT_DYNAMIC_REAL_LEGACY ? 
- FEAT_DYNAMIC ? 
- PAST_FEAT_DYNAMIC ? 

Thank you for your help!

**Comment by jaheba** [comment, marked as answer]
Hi,

`FEAT_DYNAMIC_REAL` and `FEAT_DYNAMIC_CAT` require that features are passed both in training and prediction. For example, if you want to predict demand for an item and use price as a feature, you can provide the price as a dynamic feature. Then for prediction, you need to provide the price to get a prediction.

However, if a feature is only available in training and not prediction, some models support `PAST_FEAT_DYANMIC_REAL`. For example, you could use the number of views for an item's webpage as a feature in training. For prediction, you do not pass that information, but still get a forecast.

Note while many models support `FEAT_DYNAMIC_REAL`, only few support `PAST_FEAT_DYNAMIC_REAL`. So it's best to check the specific model.

---

I think `FEAT_DYNAMIC_REAL_LEGACY` should just be removed, it's not used as far as I can see.

It appears that `FEAT_DYNAMIC` and `PAST_FEAT_DYNAMIC` are used in some models internally, so they should also ideally removed from `FieldName`. In any case, you can safely ignore these for generating data-input.
